modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -2316,22 +2377,22 @@ void ONNXImporter::parseResize(LayerParams& layerParams, const opencv_onnx::Node
     if (layerParams.get<String>("mode") == "linear" && framework_name == "pytorch")
         layerParams.set("mode", "opencv_linear");
 
-    // input = [X, scales], [X, roi, scales] or [x, roi, scales, sizes]
-    int foundScaleId = hasDynamicShapes ? node_proto.input_size() - 1
-                                        : node_proto.input_size() > 2 ? 2 : 1;
+    // opset-10: input = [X, scales]
+    // opset-11: input = [X, roi, scales] or [x, roi, scales, sizes]
+    int foundScaleId = node_proto.input_size() == 2 ? 1 : 2;
 
     Mat scales = getBlob(node_proto, foundScaleId);
```

Comment:
Won't `getBlob` fail, if it doesn't find scales blob?
The condition below seems fragile.

Reply:
scales can be empty. getBlob() doesn't fail in that case.

https://github.com/onnx/onnx/blob/master/docs/Operators.md#Resize

> The number of elements of 'scales' should be the same as the rank of input 'X'. One of 'scales' and 'sizes' MUST be specified and it is an error if both are specified. If 'sizes' is needed, the user can use an empty string as the name of 'scales' in this operator's input list.

---

" == 4" condition looks wrong. Reworked for better error messages.